Jenkinsfile

```diff
@@ -0,0 +1,396 @@
+                    export AmiId=`echo ${AmiId} | sed -e s'/"//g'`
+                    echo "Using ${AmiId}"
+                    sed -i "s/MY_AMI_ID/${AmiId}/g" InfraStack.parms.json
+                '''
+                    }
+                }
+            }
+        stage ('Prepare AWS Environment') {
+            options {
+                timeout(time: 1, unit: 'HOURS')
+            }
+            steps {
+                withCredentials(
+                    [
+                        [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: "${AwsCred}", secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'],
+                        sshUserPrivateKey(credentialsId: "${GitCred}", keyFileVariable: 'SSH_KEY_FILE', passphraseVariable: 'SSH_KEY_PASS', usernameVariable: 'SSH_KEY_USER')
+                    ]
+                ) {
+                    sh '''#!/bin/bash
+                        echo "Attempting to delete any active ${CfnStackRoot} stacks... "
+                        aws --region "${AwsRegion}" cloudformation delete-stack --stack-name "${CfnStackRoot}"
+
+                        sleep 5
+
+                        # Pause if delete is slow
+                        while [[ $(
+                                    aws cloudformation describe-stacks \
+                                      --stack-name ${CfnStackRoot} \
+                                      --query 'Stacks[].{Status:StackStatus}' \
+                                      --out text 2> /dev/null | \
+                                    grep -q DELETE_IN_PROGRESS
+                                   )$? -eq 0 ]]
+                        do
+                           echo "Waiting for stack ${CfnStackRoot} to delete..."
+                           sleep 30
+                        done
+                    '''
+                }
+            }
+        }
+        stage ('Launch Jenkins Master Stack') {
+            options {
+                timeout(time: 1, unit: 'HOURS')
+            }
+            steps {
+                withCredentials(
+                    [
+                        [$class: 'AmazonWebServicesCredentialsBinding', accessKeyVariable: 'AWS_ACCESS_KEY_ID', credentialsId: "${AwsCred}", secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'],
+                        sshUserPrivateKey(credentialsId: "${GitCred}", keyFileVariable: 'SSH_KEY_FILE', passphraseVariable: 'SSH_KEY_PASS', usernameVariable: 'SSH_KEY_USER')
+                    ]
+                ) {
+                    sh '''#!/bin/bash
+                        echo "Attempting to create stack ${CfnStackRoot}..."
+                        aws --region "${AwsRegion}" cloudformation create-stack --stack-name "${CfnStackRoot}" \
+                          --disable-rollback --capabilities CAPABILITY_NAMED_IAM \
+                          --template-url "${PrimaryTemplate}" \
+                          --parameters file://InfraStack.parms.json
```

Comment:
I'm a tad confused by this line. That you're pointing to `InfraStack.parms.json` alludes to this pipeline being meant to drive a nested CFn-template (i.e., one that deploys multiple service-elements &mdash; usually everything up to but _excluding_ the EC2 element); however, the rest of the pipeline's contents seem to indicate that this pipeline is meant to drive the `make_confluence_EC2-node.tmplt.json` CFn template. Whether driving a nested or stand-alone EC2 stack, please update the file-name to be more consistent with the pipeline's purpose.